core-web/apps/dotcms-ui/src/app/view/components/dot-device-selector/dot-device-selector.component.ts

```diff
@@ -28,9 +32,15 @@ export class DotDeviceSelectorComponent implements OnInit, OnChanges {
 
     constructor(
         private dotDevicesService: DotDevicesService,
-        private dotMessageService: DotMessageService
+        private dotMessageService: DotMessageService,
+        private readonly cd: ChangeDetectorRef
     ) {}
 
+    // Due the @input value change before change detection has been run/or completed.
+    ngAfterViewChecked() {
+        this.cd.detectChanges();
```

Comment:
Can we have a test to assert that this runs correctly?

Reply:
I removed these lines, I triggered the detectChanges at lines 79 and 84